### PR TITLE
Use cri-docker from RD when k3s version doesn't support docker API 1.44

### LINKED ISSUE #7160: Kubernetes unable to run some containers with `ImageInspectError`
### Actual Behavior

Some container images, including Rancher Desktop Kubernetes defaults, are unable to run because of an `ImageInspectError`.

### Steps to Reproduce

1. Download and install Rancher Desktop 1.14.2
2. Enable Kubernetes (wait for it to boot)
3. Run `kubectl get pods -A`

This is sufficient to reproduce on my system consistently.

### Result

```bash
kubectl get pods -A
NAMESPACE     NAME                                      READY   STATUS              RESTARTS   AGE
kube-system   coredns-6799fbcd5-bmswr                   0/1     ImageInspectError   0          63s
kube-system   local-path-provisioner-84db5d44d9-4xmlp   0/1     ImageInspectError   0          63s
kube-system   metrics-server-67c658944b-sxwst           0/1     ImageInspectError   0          63s
```

```bash
kubectl -n kube-system describe pod coredns-6799fbcd5-bmswr
Name:                 coredns-6799fbcd5-bmswr
Namespace:            kube-system
Priority:             2000000000
Priority Class Name:  system-cluster-critical
Service Account:      coredns
Node:                 lima-rancher-desktop/192.168.5.15
Start Time:           Mon, 08 Jul 2024 19:37:54 -0600
Labels:               k8s-app=kube-dns
                      pod-template-hash=6799fbcd5
Annotations:          <none>
Status:               Pending
IP:                   10.42.0.2
IPs:
  IP:           10.42.0.2
Controlled By:  ReplicaSet/coredns-6799fbcd5
Containers:
  coredns:
    Container ID:  
    Image:         rancher/mirrored-coredns-coredns:1.10.1
    Image ID:      
    Ports:         53/UDP, 53/TCP, 9153/TCP
    Host Ports:    0/UDP, 0/TCP, 0/TCP
    Args:
      -conf
      /etc/coredns/Corefile
    State:          Waiting
      Reason:       ImageInspectError
    Ready:          False
    Restart Count:  0
    Limits:
      memory:  170Mi
    Requests:
      cpu:        100m
      memory:     70Mi
    Liveness:     http-get http://:8080/health delay=60s timeout=1s period=10s #success=1 #failure=3
    Readiness:    http-get http://:8181/ready delay=0s timeout=1s period=2s #success=1 #failure=3
    Environment:  <none>
    Mounts:
      /etc/coredns from config-volume (ro)
      /etc/coredns/custom from custom-config-volume (ro)
      /var/run/secrets/kubernetes.io/serviceaccount from kube-api-access-lkq7h (ro)
Conditions:
  Type              Status
  Initialized       True 
  Ready             False 
  ContainersReady   False 
  PodScheduled      True 
Volumes:
  config-volume:
    Type:      ConfigMap (a volume populated by a ConfigMap)
    Name:      coredns
    Optional:  false
  custom-config-volume:
    Type:      ConfigMap (a volume populated by a ConfigMap)
    Name:      coredns-custom
    Optional:  true
  kube-api-access-lkq7h:
    Type:                     Projected (a volume that contains injected data from multiple sources)
    TokenExpirationSeconds:   3607
    ConfigMapName:            kube-root-ca.crt
    ConfigMapOptional:        <nil>
    DownwardAPI:              true
QoS Class:                    Burstable
Node-Selectors:               kubernetes.io/os=linux
Tolerations:                  CriticalAddonsOnly op=Exists
                              node-role.kubernetes.io/control-plane:NoSchedule op=Exists
                              node-role.kubernetes.io/master:NoSchedule op=Exists
                              node.kubernetes.io/not-ready:NoExecute op=Exists for 300s
                              node.kubernetes.io/unreachable:NoExecute op=Exists for 300s
Topology Spread Constraints:  kubernetes.io/hostname:DoNotSchedule when max skew 1 is exceeded for selector k8s-app=kube-dns
Events:
  Type     Reason         Age                   From               Message
  ----     ------         ----                  ----               -------
  Normal   Scheduled      4m3s                  default-scheduler  Successfully assigned kube-system/coredns-6799fbcd5-bmswr to lima-rancher-desktop
  Warning  Failed         116s (x12 over 4m3s)  kubelet            Error: ImageInspectError
  Warning  InspectFailed  102s (x13 over 4m3s)  kubelet            Failed to inspect image "rancher/mirrored-coredns-coredns:1.10.1": Id or size of image "rancher/mirrored-coredns-coredns:1.10.1" is not set

### Expected Behavior

Containers are able to boot and run as expected.

### Additional Information

_No response_

### Rancher Desktop Version

1.14.1

### Rancher Desktop K8s Version

1.28.4

### Which container engine are you using?

moby (docker cli)

### What operating system are you using?

macOS

### Operating System / Build Version

Sonoma 14.5

### What CPU architecture are you using?

arm64 (Apple Silicon)

### Linux only: what package format did you use to install Rancher Desktop?

None

### Windows User Only

_No response_

### PR DESCRIPTION
**Pull request by jandubois**
Updated tests can be used to verify:

```shell
RD_K3S_VERSIONS=all RD_CONTAINER_ENGINE=moby bats tests/k8s/foreach-k3s-version.bats
```

Fixes #7160

### CODE REVIEW

File: pkg/rancher-desktop/backend/backendHelper.ts [hunk 1]
```diff
@@ -143,12 +143,30 @@ export default class BackendHelper {
...
+      return true;
+    }
+    if (semver.gte(kubeVersion, '1.28.0') && semver.lte(kubeVersion, '1.28.6')) {
+      return true;
+    }
+    if (semver.gte(kubeVersion, '1.29.0') && semver.lte(kubeVersion, '1.29.1')) {
+      return true;
+    }
+
+    return false;
```
Comment (mook-as) [inline pkg/rancher-desktop/backend/backendHelper.ts:169]: This is probably easier to read when using [`semver.satisfies`](https://github.com/npm/node-semver/blob/main/functions/satisfies.js):
```suggestion
    const ranges = [
      // versions 1.24.1 to 1.24.3 don't support the --docker option
      '1.24.1 - 1.24.3',
      // cri-dockerd bundled with k3s is not compatible with docker 25.x (using API 1.44)
      // see https://github.com/k3s-io/k3s/issues/9279
      '1.26.8 - 1.26.13',
      '1.27.5 - 1.27.10',
      '1.28.0 - 1.28.6',
      '1.29.0 - 1.29.1',
    ];

    return semver.satisifies(kubeVersion, ranges.join('||'));
```
Side note: What happened to 1.25.x? They all work fine with docker API 1.44?

Comment (jandubois) [reply pkg/rancher-desktop/backend/backendHelper.ts]: > What happened to 1.25.x? They all work fine with docker API 1.44?

They work fine, but I don't know if they use API 1.44. I assume they bundle an older cri-dockerd which somehow rejects the 1.44 API during API negotiation.

The same way all the 1.24.x versions seem to work, except for 1.24.1-1.24.3, which don't support the `--docker` version.

**Review by mook-as — APPROVED** [review]

**Comment by jandubois** [comment]
Updated test fails with 1.14.2 release:

```
$ RD_K3S_VERSIONS="1.26.13 1.26.14" RD_CONTAINER_ENGINE=moby ./bats-core/bin/bats tests/k8s/foreach-k3s-version.bats
foreach-k3s-version.bats
   ===== k8s/foreach-k3s-version =====
 ✓ factory_reset 1.26.13
 ✓ start_kubernetes 1.26.13
 ✓ wait_for_kubelet 1.26.13
 ✗ wait_for_dns 1.26.13
   (from function `try' in file tests/helpers/utils.bash, line 283,
    from function `wait_for_dns' in file tests/k8s/foreach-k3s-version.bats, line 4,
    from function `_foreach_k3s_version' in test file tests/helpers/utils.bash, line 458)
     `"$2"' failed

   -- output differs --
   expected : running
   actual   : waiting
   --
 ✓ factory_reset 1.26.14
 ✓ start_kubernetes 1.26.14
 ✓ wait_for_kubelet 1.26.14
 ✓ wait_for_dns 1.26.14

8 tests, 1 failure
```

And succeeds with CI build from this PR:

```
$ RD_K3S_VERSIONS="1.26.13 1.26.14" RD_CONTAINER_ENGINE=moby ./bats-core/bin/bats tests/k8s/foreach-k3s-version.bats
foreach-k3s-version.bats
   ===== k8s/foreach-k3s-version =====
 ✓ factory_reset 1.26.13
 ✓ start_kubernetes 1.26.13
 ✓ wait_for_kubelet 1.26.13
 ✓ wait_for_dns 1.26.13
 ✓ factory_reset 1.26.14
 ✓ start_kubernetes 1.26.14
 ✓ wait_for_kubelet 1.26.14
 ✓ wait_for_dns 1.26.14

8 tests, 0 failures
```

Full test across all supported k3s version is still in progress (takes 3-4 hours).

**Comment by jandubois** [comment]
> Full test across all supported k3s version is still in progress (takes 3-4 hours).

Things run faster if we don't have to wait for the failing tests to time out:

```
520 tests, 0 failures

real	157m16.330s
```

**Pull request merged**